lib/fluent/plugin/in_watch_process.rb

```diff
@@ -45,55 +44,123 @@ def shutdown
     end
 
     def on_timer
-      io = IO.popen(@command, 'r')
-      begin
-        io.gets
-        while result = io.gets
+      @watcher.process do |data|
+        next unless @lookup_user.nil? || @lookup_user.include?(data['user'])
+        emit_tag = tag.dup
+        filter_record(emit_tag, Fluent::Engine.now, data)
+        router.emit(emit_tag, Fluent::Engine.now, data)
+      end
+    rescue StandardError => e
+      log.error "watch_process: error has occured. #{e.message}"
+    end
+
+    module Watcher
+      def self.get_watcher(keys, command)
+        if OS.linux?
+          Linux.new(keys, command)
+        elsif OS.mac?
+          Mac.new(keys, command)
+        else
+          raise NotImplementedError, "This OS type is not supported."
```

Comment:
This may be not good since setting `keys` and `command` may enable to collect data from other OS types.
I will fix this.

Reply:
I have fixed this in https://github.com/daipom/fluent-plugin-watch-process/pull/2/commits/621b59de79e7dbb706812804daa1f84933d4c5d5